### PR TITLE
coverity: fix the RESOURCE_LEAK in peeknpoke

### PR DESCRIPTION
**Pull request by bhe4**
add the close(fd) to fix the RESOURCE_LEAK issue

Tracked-On: OAM-124667

### CODE REVIEW

File: mem_dump.c [hunk 1]
```diff
@@ -93,6 +93,9 @@ int reg_write(unsigned int target, unsigned int dataBitSize, unsigned int value)
 		}
 	} else {
 		printf("Failed opening /dev/mem file\n");
+		if (fd != -1) {
+			close(fd); // Close the file descriptor if it's not -1
+		}
```
Comment (tanujtekriwal) [inline mem_dump.c:98]: return statement should be inside the if {}

File: mem_dump.c [hunk 2]
```diff
@@ -144,6 +147,9 @@ int addr_range_dump(unsigned int target, unsigned int numOfWords)
...
```
Comment (tanujtekriwal) [inline mem_dump.c:152]: return statement should be inside the if {}

**Review by tanujtekriwal — CHANGES_REQUESTED** [review]
Also please add "Tests Done: " in the commit message as part of commit message format.

**Comment by tanujtekriwal** [comment]
> Also please add "Tests Done: " in the commit message as part of commit message format.

Please add this.

**Comment by bhe4** [comment]
Added the Tests Done, Please help review.

**Review by tanujtekriwal — APPROVED** [review]
LGTM

**Review by tanujtekriwal — APPROVED** [review]
LGTM

**Pull request merged**